### PR TITLE
KAFKA-14363; Add new `group-coordinator` module (KIP-848)

### PR DESCRIPTION
**Pull request by dajac**
Introduce new `group-coordinator` module that will host the future new group coordinator as part of KIP-848.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1222,6 +1225,41 @@ project(':metadata') {
   }
 }
 
+project(':group-coordinator') {
+  archivesBaseName = "kafka-group-coordinator"
+
+  dependencies {
+    implementation project(':server-common')
+    implementation project(':clients')
+    compileOnly libs.log4j
```
Comment (ijuma) [inline build.gradle]: Do we need this?

File: build.gradle [hunk 2]
```diff
@@ -1222,6 +1225,41 @@ project(':metadata') {
...
+    testImplementation libs.junitJupiter
+    testImplementation libs.jqwik
+    testImplementation libs.hamcrest
+    testImplementation libs.mockitoCore
+    testImplementation libs.mockitoInline
```
Comment (ijuma) [inline build.gradle]: Do we need all these test dependencies? I'd consider removing `mockitoInline` and `hamcrest`.

File: build.gradle (hunk 1 above)
Comment (dajac) [reply build.gradle]: Likely not. I fixed this.

File: build.gradle (hunk 2 above)
Comment (dajac) [reply build.gradle]: I removed all the unnecessary ones and I kept the minimal that I need for now.

File: build.gradle [hunk 3]
```diff
@@ -1222,6 +1225,42 @@ project(':metadata') {
...
 
+project(':group-coordinator') {
+  archivesBaseName = "kafka-group-coordinator"
+
+  dependencies {
+    implementation project(':server-common')
+    implementation project(':clients')
+    implementation libs.slf4jApi
+
+    testImplementation project(':clients')
```
Comment (ijuma) [inline build.gradle]: Isn't this redundant given that we already have an implementation dependency on clients? Or does gradle complain if you remove it?

Comment (dajac) [reply build.gradle]: You're right. Gradle does not complain without them. I removed them.

**Review by ijuma — APPROVED** [review]
LGTM

**Pull request merged**